packages/x-tailwindcss/src/x-tailwind-plugin/components/input/default.ts

```diff
@@ -0,0 +1,37 @@
+import { TailwindHelpers } from '../../../types';
+import { inputColors } from './colors';
+import { inputSizes } from './sizes';
+
+/**
+ * Returns the default styles for component `input`.
+ *
+ * @param helpers - The {@link TailwindHelpers} to generate CSS.
+ * @returns The {@link CssStyleOptions} for the component.
+ */
+// eslint-disable-next-line  @typescript-eslint/explicit-function-return-type
+export function inputDefault(helpers: TailwindHelpers) {
+  const { theme } = helpers;
+  return {
+    fontFamily: theme('fontFamily.main'),
+    borderWidth: theme('borderWidth.1'),
+    cursor: 'text',
+    fontWeight: theme('fontWeight.regular'),
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    gap: theme('spacing.8'),
+    paddingInlineStart: theme('spacing.16'),
+    paddingInlineEnd: theme('spacing.16'),
+    '&::placeholder': {
+      color: theme('colors.neutral.50'),
+      fontFamily: theme('fontFamily.main'),
+      fontSize: theme('fontSize.md'),
+      fontWeight: theme('fontWeight.regular')
+    },
+    '&:disabled': {
+      cursor: 'not-allowed'
+    },
+    ...inputColors(helpers).neutral,
+    ...inputSizes(helpers).md
```

Comment:
question: I've seen this in more places and now I'm not sure about the approach. If we modify the `md` styles in a customer, they won't be inherited here. Not a big issue so far, but  maybe something confusing for the developers.

Also, by destructuring below the pseudoselectors, tailwind generates 3 `.x-input` rules.

Reply:
That's on purpose, we don't want to modify the default `input` when we modify de `md` size. We are just reusing code. But in a setup, we want to modify the default and the sizes separately. It happens the same with the color variant. The default color of the component is the same color as the neutral variant. But I want to modify them separately in the setup.

We will see how to make it easier to override these default styles from setups., and surely we have to modify something, but at the moment we don't have all the information.

Reply:
Understood, thanks.